src/main/java/com/diplomaProject/StudyBe/SecurityConfiguration.java

```diff
@@ -1,50 +1,123 @@
 
 @Configuration
 @EnableWebSecurity
+
 public class SecurityConfiguration {
 
-    @Bean
-    public CorsFilter corsFilter() {
-        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
-        CorsConfiguration config = new CorsConfiguration();
-        config.addAllowedOrigin("http://localhost:3000");  // Replace with your frontend's URL
-        config.addAllowedMethod("*");
-        config.addAllowedHeader("*");
-        source.registerCorsConfiguration("/**", config);
-        return new CorsFilter(source);
-    }
+    @Autowired
+    private JwtAuthenticationFilter jwtAuthFilter;
+    @Autowired
+    private AuthenticationProvider authenticationProvider;
+
+
 
     @Bean
-    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception{
-        return http
-//                .requiresChannel(channel ->
-//                        channel.anyRequest().requiresSecure())
-                .csrf(csrf -> csrf.disable())
+
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception
+    {
+        http
+                .csrf(AbstractHttpConfigurer::disable)
                 .authorizeHttpRequests(auth ->{
-                    auth.anyRequest().permitAll();
+//                    auth.anyRequest().permitAll();
 //                    auth.requestMatchers("/auth/").permitAll();
 //                    auth.requestMatchers("/admin/").hasRole("ADMIN");
-//                    auth.requestMatchers("/user/**").hasAnyRole("ADMIN","USER");
+
 //                    auth.anyRequest().authenticated();
+                    auth.requestMatchers( "/api/v1/auth/authentication", "/api/v1/auth/register").permitAll();
+                    auth.requestMatchers(HttpMethod.OPTIONS).permitAll();
+                    auth.anyRequest().authenticated();
                 })
-//                .oauth2ResourceServer(oauth2 -> oauth2
-//                        .jwt(jwt -> jwt
-//                                .jwtAuthenticationConverter(jwtAuthenticationConverter())
-//                        )
-//                )
                 .sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
-                .build();
+                .authenticationProvider(authenticationProvider )
+                .addFilterBefore(jwtAuthFilter, UsernamePasswordAuthenticationFilter.class);
+
+        return http.build();
     }
+
+//    @Bean(name = DispatcherServletAutoConfiguration.DEFAULT_DISPATCHER_SERVLET_BEAN_NAME)
+//    public DispatcherServlet dispatcherServlet() {
+//        DispatcherServlet dispatcherServlet = new DispatcherServlet();
+//        dispatcherServlet.setDispatchOptionsRequest(true);
+//        return dispatcherServlet;
+//    }
+
+    @Bean
+    public CorsFilter corsFilter() {
+        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+        List<String> options = new LinkedList<>();
+        options.add("GET");
```

Comment:
Add all variables that are security issues to a environment file. I it is not good practice to commit secrets and server config to your code base. It can be reverse engineered and secrets can leek.